lambdas/drivingpermitcheck/src/main/java/uk/gov/di/ipv/cri/drivingpermit/api/service/dva/DvaThirdPartyDocumentGateway.java

```diff
@@ -183,10 +188,13 @@ private JWSObject preparePayload(DvaPayload dvaPayload)
         }
     }
 
-    private void validateDvaResponse(DvaResponse dvaResponse)
-            throws OAuthHttpResponseExceptionWithErrorBody {
+    private void validateDvaResponse(DvaPayload dvaPayload, DvaResponse dvaResponse)
+            throws OAuthHttpResponseExceptionWithErrorBody, NoSuchAlgorithmException {
         if (Objects.nonNull(dvaResponse.getRequestHash())) {
-            // RequestHash check
+            if (!requestHashValidator.valid(dvaPayload, dvaResponse.getRequestHash())) {
```

Comment:
Can we get some logging around this. It would be nice to be able to see
logger.info("Validating DVA request hash...")
inside if logger.error("failed to validate request hash against response value")
after if logger.info("DVA request hash validated successfully against response")

Reply:
Added pre and post logging.
If it fails, the error should spit out its own logging message via `ErrorResponse.DVA_D_HASH_VALIDATION_ERROR`.

Reply:
@MMcC-GDS I might be be being blind whats the pre log?